internal/internal_decision_state_machine.go

```diff
@@ -788,6 +820,38 @@ func (h *decisionsHelper) handleRequestCancelExternalWorkflowExecutionFailed(wor
 	}
 }
 
+func (h *decisionsHelper) signalExternalWorkflowExecution(domain, workflowID, runID, signalName string, input []byte) decisionStateMachine {
+	signalID := uuid.New()
```

Comment:
This is not going to work as it is breaking determinism.

Reply:
changed